### PR TITLE
Prompts are messages

### PR DESCRIPTION
**Pull request by raulraja**
Prompts now include the list of messages, which simplifies the overall API.
This is the first part of a couple more like this which aims to simplify the usage and maintenance of the library.

### CODE REVIEW

**Review by javipacheco — CHANGES_REQUESTED** [review]
Some suggestions and one important comment related to the serialization

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/ChatWithFunctions.kt [hunk 1]
```diff
@@ -72,62 +59,38 @@ interface ChatWithFunctions : Chat {
...
-    promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS,
   ): B =
-    prompt(
-      buildPrompt {
-        +user(
-          "${inputSerializer.descriptor.serialName}(${Json.encodeToString(inputSerializer, input)})"
+    when (outputSerializer.descriptor.kind) {
+      PrimitiveKind.STRING ->
+        promptMessage(
+          prompt = Prompt { +user(Json.encodeToString(inputSerializer, input)) },
```
[the remark this answers is not in the document]
Comment (raulraja) [reply core/src/commonMain/kotlin/com/xebia/functional/xef/llm/ChatWithFunctions.kt:66]: we should not serialize in that case; we could pass the input as is if we detect it is a primitive like String

File: openai/src/commonMain/kotlin/com/xebia/functional/xef/auto/llm/openai/OpenAIScopeExtensions.kt [hunk 2]
```diff
@@ -2,112 +2,41 @@ package com.xebia.functional.xef.auto.llm.openai
...
+  functions: List<CFunction> = emptyList()
+): List<String> = model.promptMessages(prompt, this, functions)
 
 @AiDsl
 suspend inline fun <reified A, reified B> Conversation.prompt(
   input: A,
-  model: ChatWithFunctions = OpenAI().DEFAULT_SERIALIZATION,
-  promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS,
+  model: ChatWithFunctions =
+    if (A::class == String::class) OpenAI().DEFAULT_CHAT else OpenAI().DEFAULT_SERIALIZATION
```
[the remark this answers is not in the document]
Comment (raulraja) [reply openai/src/commonMain/kotlin/com/xebia/functional/xef/auto/llm/openai/OpenAIScopeExtensions.kt:28]: good catch thanks!

File: core/src/commonMain/kotlin/com/xebia/functional/xef/prompt/PromptBuilder.kt [hunk 3]
```diff
@@ -6,6 +6,10 @@ import com.xebia.functional.xef.llm.models.chat.Role
 open class PromptBuilder {
```
[the remark this answers is not in the document]
Comment (javipacheco) [reply core/src/commonMain/kotlin/com/xebia/functional/xef/prompt/PromptBuilder.kt:6]: We will address that in a separate PR

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/ChatWithFunctions.kt [hunk 4]
```diff
@@ -72,62 +59,38 @@ interface ChatWithFunctions : Chat {
     inputSerializer: KSerializer<A>,
     outputSerializer: KSerializer<B>,
     functions: List<CFunction> = generateCFunction(outputSerializer.descriptor),
-    promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS,
```
[the remark this answers is not in the document]
Comment (javipacheco) [reply core/src/commonMain/kotlin/com/xebia/functional/xef/llm/ChatWithFunctions.kt:75]: We will address that in a separate PR

**Review by javipacheco — APPROVED** [review]
🚀

**Pull request merged**